packages/cascara/src/ui/Form/Form.js

```diff
@@ -185,20 +198,29 @@ const Form = ({
 
   return (
     <ErrorBoundary>
-      <FormProvider
-        value={{
-          data,
-          enterEditMode,
-          exitEditMode,
-          isEditable,
-          isEditing,
-          onAction,
-        }}
-        {...rest}
-      >
-        {fields}
-        <ActionBar actions={renderedActions} />
-      </FormProvider>
+      {isValidDataDisplay ? (
+        <FormProvider
+          value={{
+            data,
+            enterEditMode,
+            exitEditMode,
+            isEditable,
+            isEditing,
+            onAction,
+          }}
+          {...rest}
+        >
+          {isLoading && <FormLoading />}
+          {!isLoading && (
+            <>
+              {fields}
+              <ActionBar actions={renderedActions} />
+            </>
+          )}
```

Comment:
Take a look at Table. I think we can probably just do isLoading ? TempLoading : FormContent since they can never show at the same time.